File: automobile-insurance-frontend/src/services/PaymentService.js
import api from "../api/api";

const PaymentService = {

    getAllPayments() {

        return api.get("/payments");

    },

    getPaymentById(id) {

        return api.get(`/payments/${id}`);

    },

    makePayment(payment) {

        return api.post("/payments", payment);

    },

    approvePayment(id) {

        return api.put(`/payments/${id}/approve`);

    }

};

export default PaymentService;

File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/PaymentService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.PaymentDTO;
import com.hexaware.automobileinsurance.model.Payment;

public interface PaymentService {

    Payment makePayment(PaymentDTO dto);

    Payment getPaymentById(Integer paymentId);

    List<Payment> getAllPayments();
    
    Payment approvePayment(Integer paymentId);
}